src/common/Requirements.ts

```diff
@@ -103,60 +104,88 @@ export function WrappedPromise(
         });
 }
 
-export abstract class BaseSetup implements RequirementList {
-    public skipBaseRequirements: boolean = false;
-    public skipAdditionalRequirements: boolean = false;
-    public requirements: Requirement[];
-    public additionalRequirements: Requirement[];
+export class CommandRequirement {
```

Comment:
Minor: should we consider `CommandRequirements` (with an 's') instead? By its makeup, it's always `1..n` requirements.

Reply:
The reason I didn't make it plural was because the class already has a property called `commandRequirements`. I also wanted the developer using this class be able to something like
`let commandRequirements: CommandRequirement[] = [];`
where 's' in the name of the variable can indicate a collection.
Maybe the name `CommandRequirement` here is not a good name to start with. I can add 's' or change the name. Maybe `CommandChecks`?